lib/task_listener.js

```diff
@@ -212,24 +198,63 @@ export default class TaskListener extends EventEmitter {
     }
   }
 
+  addRunningTask(runningState) {
+    this.runningTasks.push(runningState);
+
+    // After going from an idle to a working state issue a 'working' event.
+    if (this.runningTasks.length === 1) {
+      this.emit('working', this);
+    }
+  }
+
+  removeRunningTask(runningState) {
+    let taskIndex = this.runningTasks.findIndex((runningTask) => {
+      return (runningTask.taskId === runningState.taskId) &&
+        (runningTask.runId === runningState.runId)
+    });
+
+    if (taskIndex === -1) {
+      this.runtime.log('[warning] running task removal error', {
```

Comment:
Are we should this shouldn't be `[alert-operator]`, it seems to me that this breaks an important invariant and we should know if this ever happens...

Mostly because the result might be that `idle` might not be emitted etc... And we don't know what is right to do here...